src/core/qgsvectorlayercache.cpp

```diff
@@ -267,6 +273,7 @@ void QgsVectorLayerCache::layerDeleted()
 void QgsVectorLayerCache::invalidate()
 {
   mCache.clear();
+  mFullCache = false;
```

Comment:
Yes - but not silently. It emits the "invalidated" signal which is a flag to observers to rebuild a full cache (responsively!!) if required. That's what the attribute table does.